docs/tutorial/security.md

```diff
@@ -44,41 +44,495 @@ effort are always very welcome.
+has likely not been tested.
+
+Legitimate use cases exist, but unless you know what you are doing, you should
+not enable this property.
+
+### How?
+
+```js
+// Bad
+const mainWindow = new BrowserWindow({
+  webPreferences: {
+    experimentalFeatures: true
+  }
+})
+```
+
+```js
+// Good
+const mainWindow = new BrowserWindow({})
+```
+
+
+## Do Not Use `blinkFeatures`
+
+_Recommendation is Electron's default_
+
+Blink is the name of the rendering engine behind Chromium. Similarly to
+`experimentalFeatures`, the `blinkFeatures` property allows developers to
+enable features that have been disabled by default.
+
+### Why?
+
+Generally speaking, there are likely good reasons if a feature was not enabled
+by default. Legitimate use cases for enabling specific features exist. As a
+developer, you should know exactly why you need to enable a feature, what the
+ramifications are, and how it impacts the security of your application. Under
+no circumstances should you enable features speculatively.
+
+### How?
+```js
+// Bad
+const mainWindow = new BrowserWindow({
+  webPreferences: {
+    blinkFeatures: ['ExecCommandInJavaScript']
+  }
+})
+```
+
+```js
+// Good
+const mainWindow = new BrowserWindow()
+```
+
+
+## Do Not Disable WebSecurity
+
+_Recommendation is Electron's default_
+
+You may have already guessed that disabling the `webSecurity` property on a
+renderer process ([`BrowserWindow`](browser-view),
+[`BrowserView`](browser-view), or [`WebView`](web-view)) disables crucial
+security features.
+
+Legitimate use cases for this property exist in testing cases, but generally
+speaking, `webSecurity` should never be disabled in any production application.
+
+### Why?
+
+Disabling `webSecurity` will disable the same-origin policy as well as
+implicitly setting the `allowRunningInsecureContent` property to `true`. In
+other words, it allows the execution of insecure code from different domains.
+
+### How?
+```js
+// Bad
+const mainWindow = new BrowserWindow({
+  webPreferences: {
+    webSecurity: false
+  }
+})
+
+```js
+// Good
+const mainWindow = new BrowserWindow()
+```
+
+```html
+<!-- Bad -->
+<webview disablewebsecurity src="page.html"></webview>
+
+<!-- Good -->
+<webview src="page.html"></webview>
+```
+
+
+## Do Not Use `allowpopups`
+
+_Recommendation is Electron's default_
+
+If you are using [`WebViews`](web-view), you might need the pages and scripts
+loaded in your `<webview>` tag to open new windows. The `allowpopups` attribute
+enables them to create new [`BrowserWindows`](browser-window) using the
+`window.open()` method. By default, `WebViews` are not allowed to create new
+windows.
+
+### Why?
+
+If you do not need popups, you are better off not allowing the creation of
+new [`BrowserWindows`](browser-window) by default. This follows the principle
+of the minimally required access: Websites that you do not know to need popups
+should not have the ability to create new popups.
```

Comment:
s/ by default//

s/of the minimally/of minimally/

Awkward last sentence.
"Don't let a website create new popups unless you know it needs that feature."